### PR TITLE
EN-51742: Add join support in rollup

### PR DESCRIPTION
**Pull request by chitang**
Move functions from soql—server-pg to common-pg so that they can be shared between rollup and query.

### CODE REVIEW

File: common-pg/src/main/scala/com/socrata/pg/query/DataSqlizerQuerier.scala [hunk 1]
```diff
@@ -7,17 +7,17 @@ import com.socrata.datacoordinator.id.{ColumnId, UserColumnId}
 import com.socrata.datacoordinator.truth.loader.sql.AbstractRepBasedDataSqlizer
 import com.socrata.datacoordinator.truth.sql.SqlColumnRep
 import com.socrata.datacoordinator.util.CloseableIterator
+import com.socrata.pg.query.QueryResult.QueryRuntimeError
```
Comment (sjzavaleta) [inline common-pg/src/main/scala/com/socrata/pg/query/DataSqlizerQuerier.scala:10]: I see some new imports but not any new code in this file

File: common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala [hunk 2]
```diff
@@ -337,4 +358,22 @@ object RollupManager {
      SecondaryRollupInfo(rollupInfo.name.underlying, rollupInfo.soql)
 
   def shouldMaterializeRollups(stage: LifecycleStage): Boolean = stage == LifecycleStage.Published
+
+  def collectTableNames(selects: BinaryTree[Select]): Set[String] = {
+    selects match {
+      case Compound(_, l, r) =>
+        collectTableNames(l) ++ collectTableNames(r)
+      case Leaf(select) =>
+        select.joins.foldLeft(select.from.map(_.name).filter(_ != TableName.This).toSet) { (acc, join) =>
```
Comment (sjzavaleta) [inline common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala:367]: so this only collects joined table names?

File: common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala [hunk 3]
```diff
@@ -262,21 +288,16 @@ class RollupManager(pgu: PGSecondaryUniverse[SoQLType, SoQLValue], copyInfo: Cop
...
-        ctx = sqlCtx,
-        stringLit => SqlUtils.escapeString(pgu.conn, stringLit))
+
+     val selectParamSql = QueryServerHelper.sqlize(// scalastyle:ignore method.length parameter.number cyclomatic.complexity
+       pgu,
+       copyInfo.datasetInfo,
+       soqlAnalysis,
+       None, //reqCopy: Option[String],
+       None, //rollupName: Option[RollupName],
+       false, // obfuscateId: Boolean,
```
Comment (sjzavaleta) [inline common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala:298]: It seems like this is a brand new function that's used only once, curious why you have these unused params in them from the start

**Review by sjzavaleta — APPROVED** [review]

File: common-pg/src/main/scala/com/socrata/pg/query/DataSqlizerQuerier.scala (hunk 1 above)
Comment (chitang) [reply common-pg/src/main/scala/com/socrata/pg/query/DataSqlizerQuerier.scala:10]: QueryRuntimeError isn't new.  The order of imports was changed.

File: soql-server-pg/src/main/scala/com/socrata/pg/server/QueryServer.scala [hunk 4]
```diff
@@ -546,68 +547,6 @@ class QueryServer(val dsInfo: DSInfo, val caseSensitivity: CaseSensitivity, val
...
-    ) with RowReaderQuerier[SoQLType, SoQLValue]
-  }
-
-  /**
-   * @param pgu
-   * @param analysis parsed soql
-   * @param latest
-   * @return a schema for the selected columns
-   */
-  // TODO: Handle expressions and column aliases.
```
[the remark this answers is not in the document]
Comment (chitang) [reply soql-server-pg/src/main/scala/com/socrata/pg/server/QueryServer.scala:583]: added back

File: common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala (hunk 2 above)
Comment (chitang) [reply common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala:367]: it processes join queries too but not udf.

File: common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala (hunk 3 above)
Comment (chitang) [reply common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala:298]: I added comment to explain this.  TODO: Refactor QueryServer.runQuery to use this function

**Pull request merged**